wsmaster/wsmaster-local/src/main/java/org/eclipse/che/api/local/filters/EnvironmentInitializationFilter.java

```diff
@@ -45,6 +49,7 @@ public final void doFilter(
     Subject subject = new SubjectImpl("che", "che", "dummy_token", false);
     HttpSession session = httpRequest.getSession();
     session.setAttribute("codenvy_user", subject);
+    workspaceSubjectRegistry.updateSubject(subject);
```

Comment:
Why do we need to cache subject?

Reply:
for the reasons explained [here](https://github.com/eclipse/che/pull/7243#discussion_r149786402)